ted2zim/scraper.py

```diff
@@ -1,129 +1,145 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
-"""
-Class for scraping www.TED.com.
-"""
-__title__ = "webscraper"
-__author__ = "Rashiq Ahmad"
-__license__ = "GPLv3"
-
-import os
-from os import path
-import sys
-import shutil
-import distutils.dir_util
-from datetime import datetime
-from sys import platform as _platform
-import envoy
-import subprocess
-import datetime
+# vim: ai ts=4 sts=4 et sw=4 nu
+
 import dateutil.parser
 import requests
-from time import sleep
-from bs4 import BeautifulSoup
-from urlparse import urljoin
-import utils
 import json
-from jinja2 import Environment, FileSystemLoader
-from WebVTTcreator import WebVTTcreator
-from collections import defaultdict
+import pathlib
+import jinja2
+import shutil
+import datetime
+import sys
+
+from bs4 import BeautifulSoup
+from urllib.parse import urljoin
+from time import sleep
+from os import path # to be removed soon
+
+from .utils import create_absolute_link, download_from_site, build_subtitle_pages
+from .constants import ROOT_DIR, SCRAPER, MAX_SOURCE_VIDEOS_PER_PAGE, logger
+from .WebVTTcreator import WebVTTcreator
 
 
 class Ted2Zim:
 
     # The base Url. The link gives you a grid of all TED talks.
-    BASE_URL = "https://new.ted.com/talks/browse"
+    BASE_URL = "https://ted.com/talks"
     # BeautifulSoup instance
     soup = None
     # Page count
     pages = None
     # List of links to all TED talks
     videos = []
-    # Categories of the TED talks
-    categories = [
-        "technology",
-        "entertainment",
-        "design",
-        "business",
-        "science",
-        "global issues",
-    ]
-
-    def __init__(self, transcode2webm):
-        """
-        Extract number of video pages. Generate the specific
-        video page from it and srape it.
-        """
-        self.build_dir = path.join(os.getcwd(), "..", "build")
-        self.scraper_dir = path.join(self.build_dir, "TED", "scraper")
-        self.html_dir = path.join(self.build_dir, "TED", "html")
-        self.zim_dir = path.join(self.build_dir, "TED", "zim")
-        self.meta_data_dir = path.join(self.scraper_dir, "TED.json")
-        self.templates_dir = os.path.join(
-            os.path.abspath(os.path.dirname(__file__)), "templates"
-        )
+
+    def __init__(
+        self,
+        max_videos,
+        categories,
+        output_dir,
+        transcode2webm,
+        debug,
+    ):
+
+        # output directory
+        self.output_dir = pathlib.Path(output_dir).expanduser().resolve()
+
+        if max_videos.isdigit():
+            self.max_videos = int(max_videos)
+        elif max_videos == "max":
```

Comment:
this will crash on non digit value other than max. We don't want to crash on `__init__()`.
init should just set the variables and the handling go to `run()`.
Crashes on user inputs should be taken care of and a proper message returned (inside `run()`)

Reply:
Doesn't hold anymore. This has been removed now.